builder/common/types.go

```diff
@@ -597,40 +616,88 @@ func (v *VersionedSubmitBlockRequestWithProofs) String() string {
 
 // SignedConstraintsList are a list of proposer constraints that a builder must satisfy
 // in order to produce a valid bid. This is not defined on the
-// [spec](https://chainbound.github.io/bolt-docs/api/builder-api)
+// [spec](https://chainbound.github.io/bolt-docs/api/builder)
 // but it's useful as an helper type
 type SignedConstraintsList = []*SignedConstraints
 
-// Reference: https://chainbound.github.io/bolt-docs/api/builder-api
+// Reference: https://chainbound.github.io/bolt-docs/api/builder
 type SignedConstraints struct {
-	Message   ConstraintMessage   `json:"message"`
+	Message   ConstraintsMessage  `json:"message"`
 	Signature phase0.BLSSignature `json:"signature"`
 }
 
-// Reference: https://chainbound.github.io/bolt-docs/api/builder-api
-type ConstraintMessage struct {
-	Constraints    []*Constraint `json:"constraints"`
-	ValidatorIndex uint64        `json:"validator_index"`
-	Slot           uint64        `json:"slot"`
+func (c *SignedConstraints) Digest() []byte {
+	hasher := sha256.New()
+	// NOTE: ignoring errors here
+	slotBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(slotBytes, c.Message.Slot)
+
+	var top byte
+	if c.Message.Top {
+		top = 1
+	} else {
+		top = 0
+	}
+
+	hasher.Write(c.Message.Pubkey[:])
+	hasher.Write(slotBytes)
+	hasher.Write([]byte{top})
+
+	for _, tx := range c.Message.Transactions {
+		txHash := sha256.Sum256(*tx)
+		hasher.Write(txHash[:])
+	}
+
+	return hasher.Sum(nil)
+}
+
+func (c *SignedConstraints) VerifySignature(pubkey phase0.BLSPubKey) (bool, error) {
+	signingData := phase0.SigningData{ObjectRoot: phase0.Root(c.Digest()), Domain: phase0.Domain([]byte{109, 109, 111, 67})}
```

Comment:
can we add a comment explaining the COMMIT_BOOST_DOMAIN?
maybe saving it into a variable as well would be neat

Reply:
Added a bit more explanation + fixes in latest commits. Will document also in the API docs!